### PR TITLE
LUN-454: User vocabulary image thumbnail

### PR DESCRIPTION
**Pull request by f1sh1918**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

- Removed unneeded getImages fkt
- instead moved image files from temp folder to destination folder
- prefixed uri for local image files
- update podfile.lock, since was missing after react native update

### CODE REVIEW

File: src/services/helpers.ts [hunk 1]
```diff
@@ -206,16 +204,3 @@ export const getSortedAndFilteredVocabularyItems = (
 
 export const willNextExerciseUnlock = (previousScore: number | undefined, score: number): boolean =>
   score > SCORE_THRESHOLD_UNLOCK && (previousScore ?? 0) <= SCORE_THRESHOLD_UNLOCK
-
```
Comment (f1sh1918) [inline src/services/helpers.ts:209]: i'm not sure why we reading here a text file with the reference to the cached image (which may be deleted meanwhile)

Comment (SteffiStuffel) [reply src/services/helpers.ts:209]: This function is needed, as we need to load the images, for e.g. showing the VocabularyListItem  or any other component correctly. Own vocabulary should be possible to be passed and handled like lunes vocab or anything else.
This causes an error, e.g. if you open the list of own vocabulary:  `ERROR  TypeError: undefined is not an object (evaluating 'images.length')`

**Review by SteffiStuffel — CHANGES_REQUESTED** [review]

File: src/routes/process-user-vocabulary/UserVocabularyProcessScreen.tsx [hunk 2]
```diff
@@ -94,8 +94,8 @@ const UserVocabularyProcessScreen = ({ navigation }: UserVocabularyProcessScreen
 
       const imagePaths = await Promise.all(
         images.map(async (image, index) => {
-          const path = `${DocumentDirectoryPath}/image-${id}-${index}.txt`
-          await writeFile(path, image, 'utf8')
+          const path = `file:///${DocumentDirectoryPath}/image-${id}-${index}.jpg`
+          await moveFile(image, path)
```
Comment (SteffiStuffel) [inline src/routes/process-user-vocabulary/UserVocabularyProcessScreen.tsx:98]: File should not be moved, as the file should only be cached, when the the camera overlay "saves" = caches it. So if a user cancels the creation of a vocabulary no image files that do not belong to a saved images are there. Only if a user vocabulary is saved, the images should be saved.

File: src/services/helpers.ts (hunk 1 above)
Comment (f1sh1918) [reply src/services/helpers.ts:209]: can not reproduce this error. tested on real devices

File: src/routes/process-user-vocabulary/UserVocabularyProcessScreen.tsx (hunk 2 above)
Comment (f1sh1918) [reply src/routes/process-user-vocabulary/UserVocabularyProcessScreen.tsx:98]: the file will be cached and only moved if you save the vocabulary.

**Review by steffenkleinle — APPROVED** [review]
Tested on android, looks good to me :+1:

File: src/services/helpers.ts (hunk 1 above)
Comment (steffenkleinle) [reply src/services/helpers.ts:209]: Didn't see it on android as well

Comment (SteffiStuffel) [reply src/services/helpers.ts:209]: My bad, needed to clear async storage.

**Review by SteffiStuffel — APPROVED** [review]
Good work, thanks for fixing this :+1:

**Pull request merged**